fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java

```diff
@@ -4248,6 +4251,38 @@ private void createHiveTable(Database db, CreateTableStmt stmt) throws DdlExcept
         LOG.info("successfully create table[{}-{}]", tableName, tableId);
     }
 
+    private void createIcebergTable(Database db, CreateTableStmt stmt) throws DdlException {
```

Comment:
nit: 
There is a garbage collection mechanism in the `CatalogRecycleBin` class.
We may consider how data and metadata should be processed after Drop Table. Is it possible to recover through the 'recover Table'.

Reply:
@openinx
Thanks for your comment!
I could understand your context. `IcebergTable` is a kind of external_table in StarRocks.
I mean drop table should not clean up data files belonging to users. StarRocks only supports reading for hive_external_table.

I think we may focus on several points in the follow-up:
- Whether to clean up the org.apache.Table immediately after dropping the com.starrocks.IcebergTable.
- Whether to clean up the org.apache.Table only clean up metadata, such as metadata/manifest list/manifest files.
- Is it possible to support the recovery of com.starrocks.IcebergTable by `recover table <table_name>` in starrocks.
- If we support recover table, do we need to support keeping multiple points with the same table name, for example https://docs.snowflake.com/en/user-guide/data-time-travel.html#dropping-and-restoring-objects.
At present, only one table of the same name in StarRocks is reserved. Pls see
https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java#L102

These issues can be considered in subsequent iterations.

Reply:
Thanks for your comment!
We will discuss this issue later based on your suggestions.

Reply:
we should not consider drop table in iceberg since we just support query data stored in iceberg